packages/runtime/src/public/types.ts

```diff
@@ -1,3 +1,16 @@
+// Types for tab configuration
+export interface TabOptions {
+  icon: (props: { focused: boolean; color: string; size: number }) => React.ReactNode
+  label: (t: (key: string) => string) => string
+  testID?: string
+}
+
+export interface TabScreenConfig {
+  name: string // Just a unique identifier for the screen
+  component: React.ComponentType<any>
```

Comment:
This is indeed basically the pattern that I had envisioned to allow for builders to override screens in the app, but as it stands, how would a builder _usefully_ override a screen? They can pass in a custom component, but they don't really have the facilities available to _do_ anything useful within that component at the moment, no? For example, navigating to other screens within the app, dispatching actions, creating and submitting TXs. Will we expose/export those sorts of facilities to allow for users to create meaningful custom screens within the app?

Reply:
Exactly, the needed features like navigation and data access, will be added in future PRs.